http-server/src/main/java/org/triplea/server/moderator/toolbox/banned/users/UserBanService.java

```diff
@@ -37,10 +37,18 @@
   }
 
   boolean removeUserBan(final int moderatorId, final String banId) {
-    final String unbanName = bannedUserDao.lookupUserNameByBanId(banId);
+    final String unbanName = bannedUserDao.lookupUsernameByBanId(banId).orElse(null);
     if (bannedUserDao.removeBan(banId) != 1) {
       return false;
     }
+    if (unbanName == null) {
+      throw new AssertionError(
+          "Consistency error, unbanned "
+              + banId
+              + ", but "
+              + "there was no matching name for that ban.");
+    }
```

Comment:
It's a consistency check.  To summarize a bit the flow:
1. lookup username of ban
2. remove the ban (if no ban is removed, then return false)
3. update audit log, which requires the name of the banned user

Why return false instead of failing if we do not actually remove a ban? I was thinking a few reasons:
- duplicate network requests
- concurrent modification. There is no guarantee the ban list in toolbox will be 100% latest, either the same moderator with multiple toolboxes or multiple moderators each unbanning could cause this
- the outcome of this method is valid even if we do not remove a ban. We would not remove a ban when the ban does not exist. The expected outcome of this method is for the ban to not exist when we exit. Hence, it's not really an error condition.

On the other hand, if we do remove a ban, then we 100% expect to have retrieved a name on step #1, so if we remove a ban, but there was no name attached to it, we are in an unexpected/illegal state. It might make sense to use IllegalStateException perhaps.

Reply:
Ok, alternative suggestion then:
Move the username lookup after the if and use orElseThrow instead of null + null check, this way the variable is closer to the actual usage.
And yes, illegalstateexception might be a little bit more accurate. AssertionError is only used for "the JVM is broken" kind of exceptions elsewhere

Reply:
Updated the exception type: a02763b39

The if statement is deleting the ban record, perhaps why the code is a bit odd. So we need to be sure to do the lookup before the 'if' statement.